From 756c5222d698f143142b69318d5af22a377df0a0 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Mon, 15 Nov 2021 11:51:11 +0100
Subject: [PATCH 1/3] policy-node: Remove direction check

Remove the direction check in findDefinedTarget. We already to a
better direction check in canLink().

This fixes the case where a capture stream has the node.target set
to a sink. The directions don't match and the capture stream is not
linked to the monitor ports of the sink. Removing the check triggers
the right logic in canLink() and obs studio can link to monitor
ports again.
---
 src/scripts/policy-node.lua | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/scripts/policy-node.lua b/src/scripts/policy-node.lua
index b2e1e14..5029d1b 100644
--- a/src/scripts/policy-node.lua
+++ b/src/scripts/policy-node.lua
@@ -237,7 +237,6 @@ function findDefinedTarget (properties)
   local target_id = metadata
       and metadata:find(properties["node.id"], "target.node")
       or properties["node.target"]
-  local target_direction = getTargetDirection(properties)
 
   if target_id and tonumber(target_id) then
     local si_target = linkables_om:lookup {
@@ -253,7 +252,6 @@ function findDefinedTarget (properties)
       local target_props = si_target.properties
       if (target_props["node.name"] == target_id or
           target_props["object.path"] == target_id) and
-          target_props["item.node.direction"] == target_direction and
           canLink (properties, si_target) then
         return si_target
       end
-- 
2.31.1

